src/hotspot/os_cpu/bsd_aarch64/os_bsd_aarch64.hpp

```diff
@@ -40,4 +40,31 @@
     *(jlong *) dst = *(const jlong *) src;
   }
 
+  #ifdef __APPLE__
+  enum WXMode {
+    WXUnknown,
+    WXWrite,
+    WXExec
+  };
+
+  // Enables write or execute access to writeable and executable pages.
+  // returns the previous state
+  static WXMode current_thread_change_wx(WXMode new_state);
+
+  // initializes the WXMode to WXWrite, as writeable pages are the default here
+  static void current_thread_init_wx();
+
+  static void current_thread_assert_wx_state(WXMode expected);
```

Comment:
I'm open for suggestions, but putting it there was the simplest way. The problem is that os is not a namespace, but a class. But this could and should probably be changed.

Reply:
I was suggesting pushing everything in to os::ThreadWXEnable class.

Reply:
I don't think that this would be good as this would lead to fairly long calls. I would rather create a class wx and place everything in this (renaming ThreadWXEnable to Enable).